sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -391,19 +393,38 @@ def fetch_20newsgroups_vectorized(*, subset="train", remove=(), data_home=None,
 
         .. versionadded:: 0.22
 
+    as_frame : bool, default=False
+        If True, the data is a pandas DataFrame including columns with
+        appropriate dtypes (numeric, string, or categorical). The target is
+        a pandas DataFrame or Series depending on the number of
+        `target_columns`.
+
+        .. note::
+            Since the dataset is sparse, loading it as a pandas DataFrame
+            requires pandas 0.25+.
+
+        .. versionadded:: 0.24
+
     Returns
     -------
     bunch : :class:`~sklearn.utils.Bunch`
         Dictionary-like object, with the following attributes.
 
-        data: sparse matrix of shape (n_samples, n_features)
-            The data matrix to learn.
-        target: ndarray of shape (n_samples,)
-            The target labels.
+        data: {sparse matrix, dataframe} of shape (n_samples, n_features)
+            The data matrix to learn. If ``as_frame`` is `True`, ``data`` is
+            a pandas DataFrame.
+        target: {ndarray, dataframe} of shape (n_samples,)
+            The target labels. If ``as_frame`` is `True`, ``target`` is a
+            pandas DataFrame.
         target_names: list of shape (n_classes,)
             The names of target classes.
         DESCR: str
             The full description of the dataset.
+        frame: dataframe of shape (n_samples, n_features + 1)
+            Only present when `as_frame=True`. DataFrame with ``data`` and
+            ``target``.
+
+            .. versionadded:: 0.24
 
     (data, target) : tuple if ``return_X_y`` is True
```

Comment:
not sure if we need to put the information about them being a `datafram` here if `as_frame=True`

Reply:
I think that we refer to the definition of `data` and `target` previously stated which specify the format of these two variables. Here the information is more about the tuple. I think that we could be more explicit mentioning that data and target are from the above type.